client/lib/analytics/test/survicate.js

```diff
@@ -362,5 +384,62 @@ describe( 'survicate', () => {
 			addSurvicate();
 			expect( document.createElement ).not.toHaveBeenCalled();
 		} );
+
+		test( 'should set visitor traits when user is not on anonymous paths', () => {
+			const mockUser = {
+				email: 'test@example.com',
+			};
+			getCurrentUser.mockReturnValue( mockUser );
+
+			// Set non-anonymous path
+			Object.defineProperty( window, 'location', {
+				value: { pathname: '/home' },
+				writable: true,
+			} );
+
+			global._sva = {
+				setVisitorTraits: jest.fn(),
+			};
+
+			addSurvicate();
+
+			// Simulate script load
+			mockScript.onload();
+
+			expect( global._sva.setVisitorTraits ).toHaveBeenCalledWith( {
+				email: 'test@example.com',
+			} );
+		} );
+
+		test( 'should retry setting visitor traits when script is already loaded', () => {
+			const mockUser = {
+				email: 'test@example.com',
+			};
+			getCurrentUser.mockReturnValue( mockUser );
+
+			Object.defineProperty( window, 'location', {
+				value: { pathname: '/home' },
+				writable: true,
+			} );
+
+			global._sva = {
+				setVisitorTraits: jest.fn(),
+			};
+
+			// First call - loads script
+			addSurvicate();
+			mockScript.onload();
+
+			// Clear the mock to track subsequent calls
+			global._sva.setVisitorTraits.mockClear();
+
+			// Second call - should retry setting visitor traits without loading script again
+			addSurvicate();
```

Comment:
How do you ensure it doesn't load the script again? Should we add an assertion for that?